litgpt/__main__.py

```diff
@@ -0,0 +1,137 @@
+# Copyright Lightning AI. Licensed under the Apache License 2.0, see LICENSE file.
+
+from typing import TYPE_CHECKING, Any
+
+import torch
+
+if TYPE_CHECKING:
+    from jsonargparse import ArgumentParser
+
+
+def _new_parser(**kwargs: Any) -> "ArgumentParser":
+    from jsonargparse import ArgumentParser, ActionConfigFile
+
+    kwargs.setdefault("default_env", True)
+    parser = ArgumentParser(**kwargs)
+    parser.add_argument(
+        "-c", "--config", action=ActionConfigFile, help="Path to a configuration file in json or yaml format."
+    )
+    return parser
+
+
+def main() -> None:
+    from litgpt.pretrain import setup as pretrain_fn
+    from litgpt.chat.base import main as chat_fn
+    from litgpt.scripts.download import download_from_hub as download_fn
+    from litgpt.scripts.merge_lora import merge_lora as merge_lora_fn
+    from litgpt.finetune.lora import setup as finetune_lora_fn
+    from litgpt.finetune.full import setup as finetune_full_fn
+    from litgpt.finetune.adapter import setup as finetune_adapter_fn
+    from litgpt.finetune.adapter_v2 import setup as finetune_adapter_v2_fn
+    from litgpt.generate.base import main as generate_base_fn
+    from litgpt.generate.full import main as generate_full_fn
+    from litgpt.generate.lora import main as generate_lora_fn
+    from litgpt.generate.adapter import main as generate_adapter_fn
+    from litgpt.generate.adapter_v2 import main as generate_adapter_v2_fn
+    from litgpt.generate.sequentially import main as generate_sequentially_fn
+    from litgpt.generate.tp import main as generate_tp_fn
+    from litgpt.scripts.convert_pretrained_checkpoint import (
+        convert_pretrained_checkpoint as convert_pretrained_checkpoint_fn,
+    )
+    from litgpt.scripts.convert_hf_checkpoint import convert_hf_checkpoint as convert_hf_checkpoint_fn
+    from litgpt.scripts.convert_lit_checkpoint import convert_lit_checkpoint as convert_lit_checkpoint_fn
+
+    parser_data = {
+        "pretrain": {"help": "Pretrain a model.", "fn": pretrain_fn},
```

Comment:
Do the commands appear in the order given in this dict? If we can control the order, I would vote for:

download
chat
finetune
pretrain
everything else

Reply:
THey do. I'll reorder